### PR TITLE
Connect upload batch container to db

### PR DESCRIPTION
**Pull request by mckinsel**

### CODE REVIEW

File: backend/corpora/dataset_processing/process.py [hunk 1]
```diff
@@ -9,12 +9,30 @@
...
+
+except ImportError:
+    # We're in the container
+    sys.path.append("/code")
+    from common.entities.dataset import Dataset
+    from ..common.corpora_orm import DatasetArtifactFileType, DatasetArtifactType
+
+# This is unfortunate, but this information doesn't appear to live anywhere
+# accessible to the uploader
+DEPLOYMENT_STAGE_TO_URL = {
```
Comment (Bento007) [inline backend/corpora/dataset_processing/process.py:24]: Can we put this in the corpora/${stage}/config

Comment (Bento007) [reply backend/corpora/dataset_processing/process.py:24]: or it could be passed to the container by the step function.

Comment (Bento007) [reply backend/corpora/dataset_processing/process.py:24]: but this works for now.

File: backend/corpora/dataset_processing/process.py [hunk 2]
```diff
@@ -87,14 +122,21 @@ def extract_metadata(filename):
         numerator += numpy.count_nonzero(chunk)
         denominator += chunk.shape[0]
 
+    def get_term_pairs(base_term):
```
Comment (Bento007) [inline backend/corpora/dataset_processing/process.py:125]:
```suggestion
    def _get_term_pairs(base_term):
```

Comment (Bento007) [reply backend/corpora/dataset_processing/process.py:125]: and apply to all places it's used.

File: backend/corpora/dataset_processing/process.py [hunk 3]
```diff
@@ -46,6 +64,23 @@ def fix_dropbox_url(url):
     return pr.geturl()
 
 
+def update_db(metadata=None, processing_status=None):
```
Comment (Bento007) [inline backend/corpora/dataset_processing/process.py:67]: you'll need to wrap this in @db_session to make sure the changes are committed or rolled back.

Comment (Bento007) [reply backend/corpora/dataset_processing/process.py:67]: https://github.com/chanzuckerberg/corpora-data-portal/blob/7b3255bfe5c075de4da6c8ea65defa5111ae67b0/backend/corpora/common/utils/db_utils.py#L71

File: backend/corpora/dataset_processing/process.py [hunk 4]
```diff
@@ -46,6 +64,23 @@ def fix_dropbox_url(url):
...
+
+    dataset = Dataset.get(os.environ["DATASET_ID"])
+
+    if metadata:
+
+        # TODO: Delete this line one mean_genes_per_cell is in the db
```
Comment (Bento007) [inline backend/corpora/dataset_processing/process.py:73]: What does this mean?

Comment (Bento007) [reply backend/corpora/dataset_processing/process.py:73]:
```suggestion
        # TODO: Delete this line once mean_genes_per_cell is in the db
```

File: tests/unit/backend/corpora/dataset_processing/test_process.py [hunk 5]
```diff
@@ -28,38 +39,127 @@ def test_extract_metadata(self, mock_read_h5ad):
...
+            extracted_metadata["development_stage"],
+            [{lab: "adult", ont: "HsapDv:0"}, {lab: "baby", ont: "HsapDv:1"}, {lab: "tween", ont: "HsapDv:2"}],
+            self.assertDictEqual,
+        )
         self.assertEqual(extracted_metadata["cell_count"], 50001)
         self.assertAlmostEqual(extracted_metadata["mean_genes_per_cell"], numpy.count_nonzero(df) / 50001)
 
+    def test_update_db(self):
+
+        collection = Collection.create(visibility=CollectionVisibility.PUBLIC)
```
Comment (Bento007) [inline tests/unit/backend/corpora/dataset_processing/test_process.py:132]: We should only update Private collections, but this check is handled earlier in the upload process. We should change to PRIVATE for consistency.

File: tests/unit/backend/corpora/dataset_processing/test_process.py [hunk 6]
```diff
@@ -28,38 +39,127 @@ def test_extract_metadata(self, mock_read_h5ad):
...
+        dataset = Dataset.create(collection_id=collection.id, collection_visibility=CollectionVisibility.PUBLIC)
+        dataset_id = dataset.id
+
+        fake_env = patch.dict(os.environ, {"DATASET_ID": dataset_id, "DEPLOYMENT_STAGE": "test"})
+        Dataset.db.session.expire_all()
+        fake_env.start()
+
+        process.update_db(metadata={"sex": ["male", "female"]})
+        Dataset.db.session.expire_all()
```
Comment (Bento007) [inline tests/unit/backend/corpora/dataset_processing/test_process.py:141]: After you wrap `update` in `db_session` you shouldn't need to expire the sessions anymore

**Review by Bento007 — CHANGES_REQUESTED** [review]

**Comment by Bento007** [comment]
Since both of our PRs' are touching the dropbox URL's we similar code in processing the dropbox URL. We should consolidate.

File: backend/corpora/dataset_processing/process.py (hunk 4 above)
Comment (mckinsel) [reply backend/corpora/dataset_processing/process.py:132]: Currently there's no column for this in the DB. There will be one, and then we should start populating it.

File: backend/corpora/dataset_processing/process.py (hunk 1 above)
Comment (mckinsel) [reply backend/corpora/dataset_processing/process.py:26]: Yeah I think I would be better to to live in some deployment config or terraform and get passed to the container.

**Review by Bento007 — APPROVED** [review]
LGTM

**Pull request merged**